react-app-drawer-component/my-app/src/AppDrawer.tsx

```diff
@@ -0,0 +1,52 @@
+import './AppDrawer.css';
+import { FaBars } from 'react-icons/fa';
+import { useState } from 'react';
+
+type DrawerProps = {
+  heading: string;
+  items: string[];
+  onSelect: (x: string) => void;
+};
+
+type ItemProps = {
```

Comment:
I prefer to put these types right above the component that uses them. Having similar code grouped together makes it easier to read and work with.